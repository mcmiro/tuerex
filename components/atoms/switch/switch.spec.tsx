import * as React from "react";
import {screen, render} from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Switch from "./";

describe("Test switch component", () => {
	it("Should pass when isToggle and onToggle props is defined", () => {
		const isToggleMock = false;
		const onToggleMock = jest.fn();

		render(<Switch data-testid="switch-test" isToggle={isToggleMock} onToggle={onToggleMock} />);

		const step = screen.queryAllByTestId("switch-test");
		expect(step).toBeDefined();
		expect(step).toMatchSnapshot();
	});

	it("Should pass when after click component isToggle is change", () => {
		const isToggleMock = false;
		const onToggleMock = jest.fn();

		render(<Switch data-testid="switch-test" isToggle={isToggleMock} onToggle={onToggleMock} />);

		const switchComponent = screen.queryByTestId("switch-test");
		switchComponent && userEvent.click(switchComponent);
		switchComponent && expect(onToggleMock).toHaveBeenCalled();
		switchComponent && expect(isToggleMock).toBe(true);
	});
});
