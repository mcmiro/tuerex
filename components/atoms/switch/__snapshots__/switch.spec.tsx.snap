// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test switch component Should pass when isToggle and onToggle props is defined 1`] = `[]`;
